### ISSUE TITLE
42Crunch imposing 3.1.0 

### ISSUE THREAD

**Issue by Tasselhoff** [42Crunch/vscode-openapi#130]
Hi, in Visual Studio Code with 42Crunch extension, it appears to now be imposing the 3.1 version for the openapi.yml root file.

My first line is:

`openapi: '3.0.2'`

and I'm getting the following error:

`String does not match the pattern of "^3\.1\.\d+(-.+)?$".

{
	"resource": "/d:/SHIB/HL7 FHIR and OpenAPI Specification/openapi.yml",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "String does not match the pattern of \"^3\\.1\\.\\d+(-.+)?$\".",
	"source": "yaml-schema: https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.1/schema.json",
	"startLineNumber": 1,
	"startColumn": 10,
	"endLineNumber": 1,
	"endColumn": 17
}`

Note that this is only for the openapi.yml file, not for any others. And if I do change it to 3.1.0 the error goes away but I can no longer use the renderer.

**Comment by ak1394** [comment]
Hi @Tasselhoff ! Can you provide a sample of your OpenAPI file? The 42Crunch extension does not support OpenAPI 3.1 yet, and does not apply 3.1 schema to any OpenAPI files. 

Do you have  "$schema" attribute on the top level of your OpenAPI file by any chance?

**Comment by Tasselhoff** [comment]
Hmm. This was working on Monday (I took Tuesday off) and it was giving the error yesterday. Now it has disappeared again, so everything is fine. Weird...

So I guess we can close it.

**Comment by petermarcoen** [comment]
I have the same problem, here is the top part of my openapi.yaml:

```
openapi: 3.0.3
info:
    version: 1.0.0
    title: xxx
    description: xxx
```

And the error VSCode gives me:

```
{
	"resource": "xxx/openapi.yaml",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "String does not match the pattern of \"^3\\.1\\.\\d+(-.+)?$\".",
	"source": "yaml-schema: https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.1/schema.json",
	"startLineNumber": 1,
	"startColumn": 10,
	"endLineNumber": 1,
	"endColumn": 15
}
```

I tried adding the jsonSchemaDialect but that didn't help either:

`jsonSchemaDialect: https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.0/schema.json`

**Comment by Tasselhoff** [comment]
I'm not sure, to be honest. It just started working again.

**Comment by ak1394** [comment]
Thanks for reporting the issue! 

I did a little bit more digging, and it looks like it's somehow caused by RedHat's YAML extension which we depend on for yaml support. It seems to be picking it from schema store at https://www.schemastore.org/api/json/catalog.json which has OAS 3.1 schema. It can be disabled by turning "YAML > Schema Store > Enable" setting to off.

I'm not sure why does it happens intermittently, and will follow it up, but for now this setting is only workaround available.

**Comment by petermarcoen** [comment]
Hi @ak1394, interesting because that link does show that versions 3.0 and 3.1 are available, but how do I specify that?

```
{
      "name": "openapi.json",
      "description": "A JSON schema for Open API documentation files",
      "fileMatch": [
        "openapi.json",
        "openapi.yml",
        "openapi.yaml"
      ],
      "url": "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.1/schema.json",
      "versions": {
        "3.0": "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.0/schema.json",
        "3.1": "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.1/schema.json"
      }
}
```

**Comment by ak1394** [comment]
I'm not quite sure myself, it's not clear how the version is selected. The problem though is that 3.1 schema uses newer JSON Schema dialect which is not supported by VS Code yet, more details here #110

**Comment by abelbraaksma** [comment]
Not sure why this was closed, the error still shows, for sure:

```text
String does not match the pattern of "^3\.1\.\d+(-.+)?$".
```

To the people who managed to fix this, how did you go about doing it? FWIW, this happens in VS 2022 too, which I believe uses this plugin.

**Comment by ak1394** [comment]
The issue is sporadic, it looks like the author closed it once its started working for him. The underlying issue in YAML extension is not yet addressed. Easiest way to fix it is to rename your openapi file so it's called something else than "openapi.yaml".

**Comment by abelbraaksma** [comment]
@ak1394, thanks. Meanwhile I found and even easier solution. VSCode actually reads spec hints, so you can start the yaml file as follows:

```
# yaml-language-server: $schema=https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.0/schema.json

openapi: 3.0.2
```

**Comment by ivancuric** [comment]
Any updates on this?

The schema posted by @abelbraaksma returns the error:
```
Schema 'schema.json' is not valid:
/definitions/Schema/properties/multipleOf/exclusiveMinimum : must be numberYAML(768)
```

**Comment by ak1394** [comment]
Nope, it still seem to be caused by YAML extension picking the incorrect schema https://github.com/redhat-developer/vscode-yaml/issues/532

**Comment by abelbraaksma** [comment]
@ivancuric did you reference the correct version? Maybe you meant 3.1? It’s worked for me for some time now.

**Comment by ivancuric** [comment]
No, I need to use 3.0.1

**Comment by alannaidon** [comment]
@ivancuric Found a solution?

**Comment by servettonga** [comment]
@alannaidon either change the version or disable Schema Store setting. That's solved mine at least.

**Comment by bolatechproducoes** [comment]
The only solution work here is suggest by @ak1394, enter in vscode config, search `YAMLSchema Store` and uncheck the box, like in this image
![Captura de tela de 2024-05-25 03-52-41](https://github.com/42Crunch/vscode-openapi/assets/85072394/19a0618f-8e48-4706-8d82-58d879fb938a)
